### PR TITLE
Fix/prefabs

### PR DESCRIPTION
**Pull request by yalania**
Main problems were in the clone and copy methods so I have modified them and change some naming so is less confusing. I need to still fixing some stuff, but as there are many components any help discovering bugs is wellcome. I have focus on .prefabs

### CODE REVIEW

File: Engine/Component/ComponentAABB.cpp [hunk 1]
```diff
@@ -129,12 +129,9 @@ bool ComponentAABB::IsEmpty() const
 	return bounding_box.Size().Length() == 0;
 }
 
-Component* ComponentAABB::Clone(bool original_prefab) const
+Component* ComponentAABB::Clone(GameObject* owner, bool /*original_prefab*/)
```
Comment (Dartemiss) [inline Engine/Component/ComponentAABB.cpp:132]: why bool parameter is commented?

Comment (yalania) [reply Engine/Component/ComponentAABB.cpp:132]: is not use, some IDEs has warning as error and not used parameter can be an error.

**Review by Dartemiss — APPROVED** [review]
GJ! Works fine now :D

**Comment by JorxSS** [comment]
image

I tried to brake it, and I couldn't, so, VERY GOOD JOB @yalania

**Review by JorxSS — APPROVED** [review]

**Pull request merged**